runtime/drivers/athena/athena.go

```diff
@@ -0,0 +1,314 @@
+		},
+		{
+			Key:    "aws_secret_access_key",
+			Secret: true,
+		},
+	},
+}
+
+type driver struct{}
+
+type configProperties struct {
+	AccessKeyID     string `mapstructure:"aws_access_key_id"`
+	SecretAccessKey string `mapstructure:"aws_secret_access_key"`
+	SessionToken    string `mapstructure:"aws_access_token"`
+}
+
+func (d driver) Open(config map[string]any, shared bool, client activity.Client, logger *zap.Logger) (drivers.Handle, error) {
+	if shared {
+		return nil, fmt.Errorf("athena driver can't be shared")
+	}
+	conf := &configProperties{}
+	err := mapstructure.Decode(config, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := &Connection{
+		config: conf,
+		logger: logger,
+	}
+	return conn, nil
+}
+
+func (d driver) Drop(config map[string]any, logger *zap.Logger) error {
+	return drivers.ErrDropNotSupported
+}
+
+func (d driver) Spec() drivers.Spec {
+	return spec
+}
+
+func (d driver) HasAnonymousSourceAccess(ctx context.Context, src drivers.Source, logger *zap.Logger) (bool, error) {
+	return false, fmt.Errorf("not implemented")
+}
+
+type sourceProperties struct {
+	SQL            string `mapstructure:"sql"`
+	OutputLocation string `mapstructure:"output_location"`
+	Region         string `mapstructure:"region"`
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return conf, nil
+}
+
+type Connection struct {
+	config *configProperties
+	logger *zap.Logger
+}
+
+var _ drivers.Handle = &Connection{}
+
+// Driver implements drivers.Connection.
+func (c *Connection) Driver() string {
+	return "athena"
+}
+
+// Config implements drivers.Connection.
+func (c *Connection) Config() map[string]any {
+	m := make(map[string]any, 0)
+	_ = mapstructure.Decode(c.config, m)
+	return m
+}
+
+// Close implements drivers.Connection.
+func (c *Connection) Close() error {
+	return nil
+}
+
+// Registry implements drivers.Connection.
+func (c *Connection) AsRegistry() (drivers.RegistryStore, bool) {
+	return nil, false
+}
+
+// Catalog implements drivers.Connection.
+func (c *Connection) AsCatalogStore(instanceID string) (drivers.CatalogStore, bool) {
+	return nil, false
+}
+
+// Repo implements drivers.Connection.
+func (c *Connection) AsRepoStore(instanceID string) (drivers.RepoStore, bool) {
+	return nil, false
+}
+
+// OLAP implements drivers.Connection.
+func (c *Connection) AsOLAP(instanceID string) (drivers.OLAPStore, bool) {
+	return nil, false
+}
+
+// Migrate implements drivers.Connection.
+func (c *Connection) Migrate(ctx context.Context) (err error) {
+	return nil
+}
+
+// MigrationStatus implements drivers.Connection.
+func (c *Connection) MigrationStatus(ctx context.Context) (current, desired int, err error) {
+	return 0, 0, nil
+}
+
+// AsObjectStore implements drivers.Connection.
+func (c *Connection) AsObjectStore() (drivers.ObjectStore, bool) {
+	return c, true
+}
```

Comment:
It would be preferable to implement this as a `SQLStore` instead of an `ObjectStore`, and having the source be a `DatabaseSource` instead of a `BucketSource`. The DuckDB `sqlstore_to_duckdb.go` transporter is equally capable of ingesting downloaded files, so I don't think it would be a big change. And then the `fromAthena` change can be reverted in the `objectstore_to_duckdb.go` transporter.

See note about implementing as a